app/src/main/java/com/valhalla/thor/data/service/AutoFreezeManager.kt

```
@@ -0,0 +1,122 @@
+                        try {
+                            val appInfo = pm.getApplicationInfo(pkg, 0)
+                            if (appInfo.enabled) {
+                                Logger.d("AutoFreezeManager", "Auto-freezing app: $pkg")
+                                val result = manageAppUseCase.setAppDisabled(pkg, true)
+                                if (result.isSuccess) {
+                                    frozenCount++
+                                } else {
+                                    Logger.e("AutoFreezeManager", "Failed to freeze $pkg: ${result.exceptionOrNull()?.message}")
+                                }
+                            }
+                        } catch (e: PackageManager.NameNotFoundException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)